src/block-components/icon/edit.js

```diff
@@ -0,0 +1,320 @@
+/*
+ * External dependencies
+ */
+import {
+	InspectorStyleControls,
+	PanelAdvancedSettings,
+	AdvancedToolbarControl,
+	ColorPaletteControl,
+	IconControl,
+	AdvancedSelectControl,
+	ImageShapeControl,
+	AdvancedRangeControl,
+	FourRangeControl,
+} from '~stackable/components'
+import { i18n } from 'stackable'
+import {
+	useAttributeEditHandlers,
+	useBlockHoverState,
+} from '~stackable/hooks'
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n'
+import {
+	ToggleControl,
+} from '@wordpress/components'
+import { Fragment } from '@wordpress/element'
+
+export const Edit = props => {
+	const {
+		hasGradient,
+		hasShape,
+		hasBackgroundShape,
+	} = props
+
+	const [ state ] = useBlockHoverState()
+
+	const {
+		getAttribute,
+		updateAttributeHandler,
+	} = useAttributeEditHandlers()
+
+	return (
+		<InspectorStyleControls>
+			<PanelAdvancedSettings
+				title={ __( 'Icon', i18n ) }
+				id="icon"
+			>
+
+				<IconControl
+					label={ __( 'Icon', i18n ) }
+					value={ getAttribute( 'icon' ) }
+					onChange={ updateAttributeHandler( 'icon' ) }
+				/>
+
+				{ hasGradient && (
+					<Fragment>
+						<AdvancedToolbarControl
+							controls={ [
+								{
+									value: '',
+									title: __( 'Single', i18n ),
+								},
+								{
+									value: 'gradient',
+									title: __( 'Gradient', i18n ),
+								},
+							] }
+							isSmall={ true }
+							fullwidth={ false }
+							attribute="iconColorType"
+							hover="all"
+						/>
+
+						<ColorPaletteControl
+							label={ getAttribute( 'iconColorType', 'desktop', state ) === 'gradient' ? sprintf( __( 'Icon Color #%s', i18n ), 1 )
+								: __( 'Icon Color', i18n ) }
+							attribute="iconColor1"
+							hover="all"
+						/>
+						{ getAttribute( 'iconColorType', 'desktop', state ) === 'gradient' && (
```

Comment:
Let's make things simpler, the `iconColorType` should not be responsive and no hover. If the color type is gradient, it should be gradient for all devices and all hover states.